test/Plutip.purs

```diff
@@ -410,3 +465,95 @@ mkCurrencySymbol mintingPolicy = do
   mp <- mintingPolicy
   cs <- liftContractAffM "Cannot get cs" $ Value.scriptCurrencySymbol mp
   pure (mp /\ cs)
+
+mustPayToPubKeyStakeAddress
+  :: forall (o :: Type) (i :: Type)
+   . PaymentPubKeyHash
+  -> Maybe StakePubKeyHash
+  -> Value
+  -> Constraints.TxConstraints i o
+mustPayToPubKeyStakeAddress pkh Nothing = Constraints.mustPayToPubKey pkh
+mustPayToPubKeyStakeAddress pkh (Just stk) =
+  Constraints.mustPayToPubKeyAddress pkh stk
+
+assertContract :: forall (r :: Row Type). String -> Boolean -> Contract r Unit
+assertContract msg cond = if cond then pure unit else liftEffect $ throw msg
+
+assertNoUtxosAtEnterpriseAddress
+  :: forall (r :: Row Type). KeyWallet -> Contract r Unit
+assertNoUtxosAtEnterpriseAddress wallet = withKeyWallet wallet $
+  assertNoUtxosAtAddress =<< liftedM "Could not get wallet address"
+    ( payPubKeyHashEnterpriseAddress
+        <$> getNetworkId
+        <*> liftedM "Could not get payment pubkeyhash" ownPaymentPubKeyHash
+    )
+
+-- | Assert the wallet has no utxos at its base address, either
+-- | because the base address is empty or there is no base address
+-- | because the wallet has no stake key.
+assertNoUtxosAtBaseAddress
+  :: forall (r :: Row Type). KeyWallet -> Contract r Unit
+assertNoUtxosAtBaseAddress wallet = withKeyWallet wallet $
+  ownStakePubKeyHash >>= traverse_ \stake ->
+    assertNoUtxosAtAddress =<< liftedM "Could not get wallet address"
+      ( payPubKeyHashBaseAddress
+          <$> getNetworkId
+          <*> liftedM "Could not get payment pubkeyhash" ownPaymentPubKeyHash
+          <*> pure stake
+      )
+
+assertNoUtxosAtAddress :: forall (r :: Row Type). Address -> Contract r Unit
+assertNoUtxosAtAddress addr = do
+  utxos <- liftedM "Could not get wallet utxos" $ map unwrap <$> utxosAt addr
+  assertContract "Expected address to not hold utxos" $ Map.isEmpty utxos
+
+-- | For each wallet, assert that there is a one-to-one correspondance
+-- | between its utxo set and its expected utxo amounts.
+assertCorrectDistribution
+  :: forall (r :: Row Type). Array (KeyWallet /\ InitialUTxO) -> Contract r Unit
+assertCorrectDistribution wallets = for_ wallets \(wallet /\ expectedAmounts) ->
+  withKeyWallet wallet do
+    addr <- liftedM "Could not get wallet address" getWalletAddress
+    utxos <- liftedM "Could not get wallet utxos" $ map unwrap <$> utxosAt addr
+    assertContract "Incorrect distribution of utxos" $
+      checkDistr utxos expectedAmounts
+  where
+  -- Idea here is to iterate through the expected amounts and remove
+  -- one matching utxo from the utxo set if found, otherwise return
+  -- false. Once we've gone through all expected amounts, if all of
+  -- them have been found in the utxo set, we expect there to be no
+  -- utxos remaining
+  checkDistr :: Utxo -> InitialUTxO -> Boolean
```

Comment:
Could you please also rename `InitialUTxO` to `InitialUTxOs`. I think this is just a mistake on my side.